### PR TITLE
GH 2054 Range Support

### PR DESCRIPTION
**Pull request by GordonSmith**

### CODE REVIEW

File: src/marshaller/HipieDDL.js [hunk 1]
```diff
@@ -88,6 +53,54 @@
...
+                this.columns.push(key);
+                this.columnsIdx[key] = this.columns.length - 1;
+            }
+            this.columnsRHS[this.columnsIdx[key]] = this.mappings[key];
+            this.columnsRHSIdx[this.mappings[key]] = this.columnsIdx[key];
+            this.hasMappings = true;
+        }
+    };
+
+    SourceMappings.prototype.init = function() {
```
Comment (bwood1) [inline src/marshaller/HipieDDL.js]: Is this meant to be a duplicate of the function above?

File: src/marshaller/HipieDDL.js [hunk 2]
```diff
@@ -912,6 +937,10 @@
                                 var selectOptions = [];
                                 var options = [];
                                 var inp;
+                                if (!field.properties.charttype && field.properties.type === "range") {
+                                    //  TODO - Verify with @DL
```
Comment (bwood1) [inline src/marshaller/HipieDDL.js]: Is this TODO done?

File: src/marshaller/HipieDDL.js [hunk 3]
```diff
@@ -1232,7 +1264,7 @@
     };
 
     Visualization.prototype.hasSelection = function () {
-        return this._widgetState && this._widgetState.selected;
+        return this._widgetState && this._widgetState.row;//  TODO TMP Testing should be:  selected;
```
Comment (bwood1) [inline src/marshaller/HipieDDL.js]: Does this need to be switched back to selected?

File: src/marshaller/HipieDDL.js (hunk 1 above)
Comment (GordonSmith) [inline src/marshaller/HipieDDL.js]: I think I just moved "hipieType2DBType" to the top of the file and this is how GitHub diff is showing it...

File: src/marshaller/HipieDDL.js (hunk 2 above)
Comment (GordonSmith) [inline src/marshaller/HipieDDL.js]: Nope

File: src/marshaller/HipieDDL.js (hunk 3 above)
Comment (GordonSmith) [inline src/marshaller/HipieDDL.js]: Yup

**Pull request merged**